src/components/Modals/AddAddress/index.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  MenuItem,
+  TextField,
+} from '@mui/material';
+import { contractTx, useInkathon } from '@scio-labs/use-inkathon';
+import { useEffect, useState } from 'react';
+
+import { useToast } from '@/contexts/Toast';
+import { useIdentity } from '@/contracts';
+import { NetworkId } from '@/contracts/types';
+
+interface AddAddressModalProps {
+  open: boolean;
+  onClose: () => void;
+}
+
+export const AddAddressModal = ({ open, onClose }: AddAddressModalProps) => {
+  const { api, activeAccount } = useInkathon();
+  const { networks, contract } = useIdentity();
+  const { toastError, toastSuccess } = useToast();
+
+  const [networkId, setNetworkId] = useState<NetworkId>(0);
```

Comment:
This shouldn't be set to zero by default. The user shouldn't be able to add an address without selecting the network. Could you add a couple of networks to the contract so that this can be properly tested?